src/AutoPollConfigService.ts

```diff
@@ -27,11 +28,14 @@ export class AutoPollConfigService extends ConfigServiceBase<AutoPollOptions> im
                 resolve();
             });
 
+            this.initialization.then(() => !this.disposed && options.hooks.emit("clientReady"));
```

Comment:
It would be nice to extract event names to constants.

Reply:
I'm not sure about this for two reasons:
1. Other libs don't do that either (at least, the ones I know).
2. Since the possible event names are encoded in [this type](https://github.com/configcat/common-js/pull/63/commits/438d659cc6ac2aa04d4d4d138d0cda57f12ae4cc#diff-9cfc433aea5cb0c5cdb66f2b712b9f25cb171f5345f4c47e831fd83850c692c5R5-R11), the TS compiler will make sure that consumers can only use existing event names and IDEs can also offer the possible event names via their "intellisense" features.

Reply:
I think it can be useful (with export) for JS (not TS) customers.

Reply:
After discussing this in detail, we decided that constants (i.e. identifiers for string constants) wouldn't add real value as on the one hand TS can typecheck string constants just as well as identifiers, on the other hand consumers who use JS directly won't really benefit from constants as in that case intellisense is pretty limited and there's no preliminary compilation step either which would check that the identifier is valid (e.g. not mistyped).